### ISSUE TITLE
jit_llvm_bundle.s marked as GENERATED, is not being re-generated when missing

### ISSUE THREAD

**Issue by mrzzzrm** [hyrise/hyrise#1328]
This is the only issue I have encountered so far when using hyrise with ninja-build. jit_llvm_bundle.s is generated as part of the cmake configuration and marked as "GENERATED". Consequently, it gets deleted when `ninja clean` is called, but it is not re-generated as part of the build, leading to the build failing.

Why generate this in the CMake configuration step in the first place? Can't we just ship it as normal source? The less happens in CMake (as this issue proves again) the better.

@FabianWiebe

**Comment by FabianWiebe** [comment]
> Can't we just ship it as normal source?

Probably yes. My first tries with moving the function `file` in `EmbedLLVM.cmake` to a `add_custom_command` function produced some linker errors but this should probably work.

**Comment by mrzzzrm** [comment]
This is the content of the file for me:

```
        .global _ZN7opossum15jit_llvm_bundleE
        .global __ZN7opossum15jit_llvm_bundleE
        .global _ZN7opossum20jit_llvm_bundle_sizeE
        .global __ZN7opossum20jit_llvm_bundle_sizeE
        .p2align 3
        _ZN7opossum15jit_llvm_bundleE:
        __ZN7opossum15jit_llvm_bundleE:
        .incbin "/home/moritz/Coding/hyrise/cmake-build-debug-clang-6/src/lib/embed_llvm/jit_llvm_bundle.bc"
        1:
        .p2align 3
        _ZN7opossum20jit_llvm_bundle_sizeE:
        __ZN7opossum20jit_llvm_bundle_sizeE:
        .8byte 1b - _ZN7opossum15jit_llvm_bundleE
```

why not change `/home/moritz/Coding/hyrise/cmake-build-debug-clang-6/src/lib/embed_llvm/jit_llvm_bundle.bc` to "src/lib/embed_llvm/jit_llvm_bundle.bc", include the asm file as normal source file and not generate anything at compile/config time at all?

**Comment by mrzzzrm** [comment]
If we absolutely need to generate this file, cmake has a command for this: https://cmake.org/cmake/help/v3.0/command/configure_file.html

**Comment by FabianWiebe** [comment]
> why not change `/home/moritz/Coding/hyrise/cmake-build-debug-clang-6/src/lib/embed_llvm/jit_llvm_bundle.bc` to "src/lib/embed_llvm/jit_llvm_bundle.bc", include the asm file as normal source file and not generate anything at compile/config time at all?

The file `.../src/lib/embed_llvm/jit_llvm_bundle.bc` is within your cmake folder. `jit_llvm_bundle.s` refers to this file so that `jit_llvm_bundle.s` cannot be a normal source file.

I will have a look at `configure_file`.

**Comment by FabianWiebe** [comment]
I am currently on it.

**Comment by FabianWiebe** [comment]
If the file is not generated during CMake config phase, the config phase fails later with:
```
CMake Error at src/lib/CMakeLists.txt:679 (add_library):
  Cannot find source file:

    /Users/fabian/git/hyrise/cmake-build-debug/src/lib/embed_llvm/jit_llvm_bundle.s

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
```
So the file `jit_llvm_bundle.s` must already be created during the CMake config phase.

@mrzzzrm Can you try this fix: 6ac69f1
The crucial part is to set the `PROPERTY` `GENERATED` to `FALSE`.

**Comment by FabianWiebe** [comment]
@mrzzzrm Feel free to take this issue over, if you want.
